src/builder/api/modifier.go

```diff
@@ -1,5 +1,6 @@
 package api
```

Comment:
I don't really like the name `api` as a package name as it is too generic imo.
I'd just move it to the builder package, still very generic, but `builder.Modifier` is clearer than `api.Modifier`

Reply:
I like having the interface in a separate package

Reply:
why

Reply:
so this can be done (without coming up with weird names) 
```
type Builder[T any] struct {
	data      *T
	modifiers []api.Modifier[T]
}

var _ api.Builder[any] = (*Builder[any])(nil)
```
so we can make a difference between the interface and the generic implementation